particles/Tutorial/Javascript/Demo/source/MoveApplier.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+ /* global defineParticle */
+
+ defineParticle(({DomParticle, html}) => {
+
+  // TODO (heimlich@) tied games
+  return class extends DomParticle {
+    update({move, gameState}, {}) {
+      // If the move is reset or start, reset the gameState
+      if (move && (move.move == 'start' || move.move == 'reset')) {
+        const gs = {
+          board: ',,,,,,,,',
+          gameOver: false,
+          winnerAvatar: null,
+          currentPlayer: 1,
+          lastMove: '',
+          gameStarted: true
+        };
+        this.updateSingleton('gameState', gs);
+      }
+      if (move && gameState && gameState.lastMove != move.move) {
+        // Get the old gameState values to update.
+        let newBoard = gameState.board;
+        let newGameOver = gameState.gameOver;
+        let newWinnerAvatar = gameState.winnerAvatar;
+        let newCurrentPlayer = gameState.currentPlayer;
+
+        // Get the board and move in a usable state
+        const arr = gameState.board.split(`,`);
+        const mv = parseInt(move.move, 10) - 1;
+        
+        // If the move is valid
+        if (arr[mv] == ``) {
+
+          // Apply move
+          arr[mv] = move.playerAvatar;
+          newBoard = arr.join();
+
+          // Define all the possible winning sequences
```

Comment:
Do you think this comment is adding value?

To me it feels uncomfortably close to:

```
/**
 * Gets the object factory.
 */
public ObjectFactory getObjectFactroy() {
  ...
}
```

I'll take a minute here to expand my thinking around in-code comments. The rules for class comments or method comments are a bit different. I hope this is non controversial:

1. You want to optimise the code for reading.
1. Comments actually take time and mental energy to read.
1. They are great if there is meaning that cannot be captured purely with good naming and good structure. Usually you would explain why your code does something or the way it does something that requires context. E.g. "We need to perform this operation this way because IE6 does not support Object.entries()" or "This algorithm is O(n^2), but that's ok because we ensure n < 20 at the callsite. A more optimal algorithm would be harder to maintain and may actually be slower for low values of n".
1. Comments that purely describe what the code is doing, e.g. "We set the value of x to 5", "We define all possible winning sequences" - when the variable name is winning sequences are adding friction during reading while they don't add value. Cutting them is like cutting useless words or sentences in your prose, which is a good thing if you write a technical thing. E.g. https://g3doc.corp.google.com/company/engedu/writing/short-sentences.md?cl=head#eliminate-or-reduce-extraneous-words